train_gpt2.cu

```diff
@@ -3020,6 +3020,18 @@ void logger_log_train(Logger *logger, int step, float train_loss) {
     }
 }
 
+void logger_log_metrics(Logger *logger, int step, float norm, float lr, float latency, float mfu, float tokens) {
+    if (logger->active == 1) {
+        FILE *logfile = fopenCheck(logger->output_log_file, "a");
+        fprintf(logfile, "s:%d norm:%6.4f\n", step, norm);
```

Comment:
At a previous unnamed company i got a bit burned with logging a zillion metrics to wandb, it bloated up files and made everything super slow, and people always just looked at the same 5 plots anyway. Do we want to record all of latency, mfu, and tok/s? Also, the variable here is `tokens` but actually it's `tokens_per_second`. These are all the same number, just converted one way or another. Hmm.

Reply:
zillion 🤯. It’s true that the project becomes bloated. Let me refactor a bit.

Reply:
@karpathy Maybe probably not a good idea to have this integration. Still somebody can have plotting mechanism which reads the output logs and produce really necessary graphs without bloating the system ☝️